PHP Deprecated, Using null as an array offset is deprecated

I attempted to install Xdebug via PIE on Windows with PHP 8.4 and saw an error:
```
C:\php>php pie.phar install xdebug/xdebug

Box Requirements Checker
========================

> Using PHP 8.4.10
> PHP is not using any php.ini file.

> Checking Box requirements:
  .......E...

 [ERROR] Your system is not ready to run the application.

Fix the following mandatory requirements:
=========================================

 * The package "composer/ca-bundle" requires the extension "openssl".
   You either need to enable it or request the application to be
   shipped with a polyfill for this extension.

C:\php>
```
I updated to PHP 8.5.3 and the error became:
```
C:\php>php pie.phar install xdebug/xdebug
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Box Requirements CheckerPHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

========================PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
> Using PHP 8.5.3PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
> PHP is using the following php.ini file:PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

  C:\php\php.iniPHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
> Checking Box requirements:PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
  .......E...PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

 [ERROR] Your system is not ready to run the application.PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Fix the following mandatory requirements:PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

=========================================PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
 * The package "composer/ca-bundle" requires the extension "openssl". You either need to enable it or
   request the application to be shipped with a polyfill for this extension.
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

C:\php>php pie.phar install xdebug/xdebug
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Box Requirements CheckerPHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

========================PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
> Using PHP 8.5.3PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
> PHP is using the following php.ini file:PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

  C:\php\php.iniPHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
> Checking Box requirements:PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
  .......E...PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

 [ERROR] Your system is not ready to run the application. PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Fix the following mandatory requirements:PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

=========================================PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
 * The package "composer/ca-bundle" requires the extension "openssl". You either need to enable it or
   request the application to be shipped with a polyfill for this extension.
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97
PHP Deprecated:  Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

Deprecated: Using null as an array offset is deprecated, use an empty string instead in phar://C:/php/pie.phar/.box/src/Printer.php on line 97

C:\php>
```
Does PIE need to be changed for PHP 8.5?

I fixed the error by copying `php.ini-development` to `php.ini` and uncommenting `extension_dir = "ext"` and `extension=openssl`. This may need to be added to the Windows install instructions.

--- a/box.json.dist
+++ b/box.json.dist
@@ -2,6 +2,7 @@
     "alias": "pie.phar",
     "output": "pie.phar",
     "git": "pie_version",
+    "stub": "phar-stub.php",
     "force-autodiscovery": true,
     "directories": [
         "resources"

--- a/phar-stub.php
+++ b/phar-stub.php
@@ -1,0 +1,19 @@
+#!/usr/bin/env php
+<?php
+
+declare(strict_types=1);
+
+// echo "James says hello :)\n";
+
+// Workaround for https://github.com/php/pie/issues/537 and https://github.com/box-project/box/issues/1577
+error_reporting(error_reporting() & ~E_DEPRECATED);
+
+Phar::mapPhar('pie.phar');
+
+require 'phar://pie.phar/.box/bin/check-requirements.php';
+
+$_SERVER['SCRIPT_FILENAME'] = 'phar://pie.phar/bin/pie';
+require 'phar://pie.phar/bin/pie';
+
+// phpcs:ignore Generic.Files.InlineHTML.Found
+__HALT_COMPILER(); ?>

